s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -58,6 +60,32 @@ private[alpakka] final class S3Stream(credentials: AWSCredentials, region: Strin
   val MinChunkSize = 5242880 //in bytes
   val signingKey = SigningKey(credentials, CredentialScope(LocalDate.now(), region, "s3"))
 
+  /** listBucket lists files in a bucket, filtered by an optional prefix 
+    * 
+    * @param bucket name of the bucket
+    * @param prefix optional prefix
+    * @return A akka Source with keys
+    * 
+    */
+  def listBucket(bucket: String, prefix: Option[String] = None): Source[String, NotUsed] = {
```

Comment:
Good question! Sounds like a design decision for the maintainers, I am not really sure what the best practice is ....